api/api.go

```diff
@@ -982,3 +996,21 @@ func requireOK(d time.Duration, resp *http.Response, e error) (time.Duration, *h
 	}
 	return d, resp, nil
 }
+
+// Context returns the context used for canceling HTTP requests related to this query
+func (o *QueryOptions) Context() context.Context {
+	if o != nil && o.ctx != nil {
+		return o.ctx
+	}
+	return context.Background()
+}
+
+// WithContext creates a copy of the query options using the provided context to cancel related HTTP requests
```

Comment:
Naming this method `WithContext` feels like it's a reference to the `context.WithContext` but it operates differently... I might be missing something but why copy the `QueryOptions` object instead of assign the passed context to the `ctx` field?

Reply:
@tgross I will implement whatever API surface you're comfortable with as maintainer.
I went with this seemingly unintuitive API because it is what Consul does
https://github.com/hashicorp/consul/blob/master/api/api.go#L178

I felt like consistency with Consul's API was a good starting place.

If you don't like it I suggest just exporting a member called `Context` on `QueryOptions` and letting callers set it like they would any other property. Please let me know how to proceed.

Reply:
Let's keep it! See my remarks in https://github.com/hashicorp/nomad/pull/8836/files#r485614197